pkgs/development/compilers/factor-lang/factor98.nix

```diff
@@ -0,0 +1,178 @@
+{ stdenv, lib, git, curl, makeWrapper, fetchurl, unzip, runCommand, writeScriptBin
+, interpreter
+, glib
+, pango
+, cairo
+, gtk2-x11
+, gdk-pixbuf
+, gnome2
+, pcre
+, libGL
+, libGLU
+, freealut
+, openssl
+, udis86
+, openal
+, libogg
+, libvorbis
+, graphviz
+, librsvg
+, zlib
+, tzdata
+, ncurses
+}:
+let
+  runtimeLibs = [ glib pango cairo gtk2-x11 gdk-pixbuf gnome2.gtkglext pcre
+    libGL libGLU freealut openssl udis86 openal libogg libvorbis graphviz zlib ];
+
+  wrapFactorSh = { from, to ? false, runtimeLibs }: ''
+    # Set Gdk pixbuf loaders file to the one from the build dependencies here
+    unset GDK_PIXBUF_MODULE_FILE
+    # Defined in gdk-pixbuf setup hook
+    findGdkPixbufLoaders "${librsvg}"
+
+    ${if to then "makeWrapper ${from} ${to}" else "wrapProgram ${from}"} \
+      --set GDK_PIXBUF_MODULE_FILE "$GDK_PIXBUF_MODULE_FILE" \
+      --argv0 factor \
+      --prefix LD_LIBRARY_PATH : /run/opengl-driver/lib:${lib.makeLibraryPath runtimeLibs} \
+      --prefix PATH : ${lib.getBin graphviz}/bin
+  '';
+
+  wrapFactor = runtimeLibs:
+    runCommand (lib.appendToName "with-libs" interpreter).name {
+      nativeBuildInputs = [ makeWrapper gdk-pixbuf ];
+      passthru.runtimeLibs = runtimeLibs ++ interpreter.runtimeLibs;
+    } (wrapFactorSh { from = "${interpreter}/lib/factor/.factor.wrapped";
+                      to = "$out/bin/factor";
+                      runtimeLibs = (runtimeLibs ++ interpreter.runtimeLibs);
+                    });
+
+  # Development helper for use in nix shell
+  wrapLocalFactor = writeScriptBin "wrapFactor" ''
+    #!${stdenv.shell}
+    ${wrapFactorSh { from = "./factor"; inherit runtimeLibs; }}
+    ln -sf factor.image .factor-wrapped.image
+  '';
+  rev = "7999e72aecc3c5bc4019d43dc4697f49678cc3b4";
+  version = "0.98";
+
+in stdenv.mkDerivation {
+  pname = "factor-lang";
+  inherit rev version;
+
+  src = fetchurl {
+    url = "https://downloads.factorcode.org/releases/${version}/factor-src-${version}.zip";
+    sha256 = "01ip9mbnar4sv60d2wcwfz62qaamdvbykxw3gbhzqa25z36vi3ri";
+  };
+
+  patches = [
+    ./staging-command-line-0.98-pre.patch
+    ./workdir-0.98-pre.patch
+    ./adjust-paths-in-unit-tests.patch
+  ];
+
+  nativeBuildInputs = [ git makeWrapper curl unzip wrapLocalFactor ];
```

Comment:
How is curl used? You are sure it does not need to go to buildInputs?

Reply:
It is there to make configure happy. I refrained from patching out the test for curl, because I build the package in a way that never tries to use curl, anyway.